apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/interceptor/ApolloRequest.kt

```diff
@@ -4,10 +4,13 @@ import com.apollographql.apollo.api.ApolloExperimental
 import com.apollographql.apollo.api.ExecutionContext
 import com.apollographql.apollo.api.Operation
 import com.apollographql.apollo.api.ScalarTypeAdapters
+import com.benasher44.uuid.uuid4
 
 @ApolloExperimental
-class ApolloRequest<T> constructor(
-    val operation: Operation<*, T, *>,
+class ApolloRequest<D : Operation.Data>(
+    val operation: Operation<D, D, *>,
     val scalarTypeAdapters: ScalarTypeAdapters,
-    val executionContext: ExecutionContext = ExecutionContext.Empty
-)
+    val executionContext: ExecutionContext
+) {
+  val requestUuid = uuid4()
```

Comment:
Looks like `requestUuid` is specific to WebSockets, a bit similar to how headers are specific to Http. Should it be in an opaque `ExecutionContext.Element`

Reply:
I would leave it to be a pat of the request `ApolloRequest` object for the future as it gives a way to match request with the corresponding response by `requestUuid`. I feel it will be helpful if someone requires it in own custom interceptor implementation.